src/config-loader.ts

```diff
@@ -22,7 +22,7 @@ export interface ConfigLoaderParams {
 export interface ConfigLoaderSuccessResult {
   resultType: "success";
   configFileAbsolutePath: string;
-  baseUrl: string;
+  baseUrl?: string;
```

Comment:
This is why it's a breaking change

Reply:
Making a previously required property optional is not a breaking change IMO, especially in this case:

* It won't change anything for projects having a `baseUrl`
* There's virtually no projects using `tsconfig-paths` that currently do not have a `baseUrl`, because it was basically broken

Reply:
This PR introduces a compiler error into any downstream package that happens to use `loadConfig` and expects the `baseUrl` property to only ever be a string. If a change can cause downstream compiler errors, I consider it breaking.

Reply:
Oh I was not aware of the public API, my bad!